lib/map/index.d.ts

```diff
@@ -0,0 +1,22 @@
+interface IterableIterator<T> {}
+interface Iterable<T> {}
+
+interface Map<K, V> {
+  clear(): void;
+  delete(key: K): boolean;
+  entries(): IterableIterator<[K, V]>;
+  forEach(callbackfn: (value: V, index: K, map: Map<K, V>) => void, thisArg?: any): void;
+  get(key: K): V;
+  has(key: K): boolean;
+  keys(): IterableIterator<K>;
+  set(key: K, value?: V): Map<K, V>;
+  size: number;
+  values(): IterableIterator<V>;
+}
+
+interface MapConstructor {
+  new <K, V>(): Map<K, V>;
+  new <K, V>(iterable: Iterable<[K, V]>): Map<K, V>;
```

Comment:
We should probably delete this constructor definition `new <K, V>(iterable: Iterable<[K, V]>): Map<K, V>;` as instanciating from an iterable is not supported in IE11. Valid also for the `keys`, `entries` and `values` method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map